cmd/client/commandv2/create/createhttpproxy.go

```diff
@@ -0,0 +1,395 @@
+ */
+
+package create
+
+import (
+	"encoding/base64"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/megaease/easegress/v2/cmd/client/general"
+	"github.com/megaease/easegress/v2/cmd/client/resources"
+	"github.com/megaease/easegress/v2/pkg/filters"
+	"github.com/megaease/easegress/v2/pkg/filters/proxies"
+	"github.com/megaease/easegress/v2/pkg/filters/proxies/httpproxy"
+	"github.com/megaease/easegress/v2/pkg/object/httpserver"
+	"github.com/megaease/easegress/v2/pkg/object/httpserver/routers"
+	"github.com/megaease/easegress/v2/pkg/object/pipeline"
+	"github.com/megaease/easegress/v2/pkg/util/codectool"
+	"github.com/spf13/cobra"
+)
+
+// HTTPProxyOptions are the options to create a HTTPProxy.
+type HTTPProxyOptions struct {
+	Name  string
+	Port  int
+	Rules []string
+
+	TLS        bool
+	AutoCert   bool
+	CaCertFile string
+	CertFiles  []string
+	KeyFiles   []string
+
+	caCert string
+	certs  []string
+	keys   []string
+	rules  []*HTTPProxyRule
+}
+
+var httpProxyOptions = &HTTPProxyOptions{}
+
+var httpProxyExamples = `# General case
+egctl create httpproxy NAME --port PORT \ 
+	--rule HOST/PATH=ENDPOINT1,ENDPOINT2 \
+	[--rule HOST/PATH=ENDPOINT1,ENDPOINT2] \
+	[--tls] \
+	[--auto-cert] \
+	[--ca-cert-file CA_CERT_FILE] \
+	[--cert-file CERT_FILE] \
+	[--key-file KEY_FILE]
+
+# Create a HTTPServer (with port 10080) and corresponding Pipelines to direct 
+# request with path "/bar" to "http://127.0.0.1:8080" and "http://127.0.0.1:8081" and 
+# request with path "/foo" to "http://127.0.0.1:8082".
+egctl create httpproxy demo --port 10080 \
+	--rule="/bar=http://127.0.0.1:8080,http://127.0.0.1:8081" \
+	--rule="/foo=http://127.0.0.1:8082"
+
+# Create a HTTPServer (with port 10081) and corresponding Pipelines to direct request 
+# with path prefix "foo.com/prefix" to "http://127.0.0.1:8083".
+egctl create httpproxy demo2 --port 10081 \
+	--rule="foo.com/prefix*=http://127.0.0.1:8083"
+`
+
+// HTTPProxyCmd returns create command of HTTPProxy.
+func HTTPProxyCmd() *cobra.Command {
+	o := httpProxyOptions
+
+	cmd := &cobra.Command{
+		Use:     "httpproxy NAME",
+		Short:   "Create a HTTPServer and corresponding Pipelines with a specific name",
+		Args:    httpProxyArgs,
+		Example: general.CreateMultiLineExample(httpProxyExamples),
+		Run: func(cmd *cobra.Command, args []string) {
+			err := httpProxyRun(cmd, args)
+			if err != nil {
+				general.ExitWithError(err)
+			}
+		},
+	}
+
+	cmd.Flags().IntVar(&o.Port, "port", -1, "Port of HTTPServer")
+	cmd.Flags().StringArrayVar(&o.Rules, "rule", []string{}, "Rule in format host/path=endpoint1,endpoint2. Paths containing the leading character '*' are considered as PathPrefix.")
+
+	cmd.Flags().BoolVar(&o.TLS, "tls", false, "Enable TLS")
+	cmd.Flags().BoolVar(&o.AutoCert, "auto-cert", false, "Enable auto cert")
+	cmd.Flags().StringVar(&o.CaCertFile, "ca-cert-file", "", "CA cert file")
+	cmd.Flags().StringArrayVar(&o.CertFiles, "cert-file", []string{}, "Cert file")
```

Comment:
it use as `--cert-file filename1 --cert-file filename2`... If use `--cert-files` it feels more like `--cert-files filename1,filename2,filename3`, and it should use `cmd.Flags().StringSliceVar`. But it may cause some problem, since filename may itself contains `,`.